core/src/ldml/C7043_ldml.md

```diff
@@ -324,10 +324,115 @@ For each key:
+|20+|  32  | target     | int: target vkey ID (0…255)              |
+|28+|  32  | layer      | int: index to first layer element        |
+|32+|  32  | layerCount | int: number of `layr` elements in list   |
+|36+|  96  | reserved   | short`[3]`: padding                      |
+
+- `flags`: a 32-bit bitfield defined as below:
+
+  | Bit position | Meaning  |  Description         |
+  |--------------|----------|----------------------|
+  |       0      | form     | 0: hardware          |
+  |       0      | form     | 1: touch             |
+
+For each layer element
+
+| ∆ | Bits | Name       | Description                              |
+|---|------|------------|------------------------------------------|
+|16+|  32  | id         | str: layer id such as `base` or `shift`  |
+|24+|  32  | modifier   | str: modifier string                     |
+|28+|  32  | row        | int: `rows` identifier of first row      |
+|32+|  32  | rowCount   | int: number of `rows` elements           |
+
+### C7043.2.x `rows`—Row list
+
+Each list is a row of keys in the `key2` element
+
+| ∆ | Bits | Name    | Description                              |
+|---|------|---------|------------------------------------------|
+| 0 |  32  | ident   | `rows`                                   |
+| 4 |  32  | size    | int: Length of section                   |
+| 8 |  32  | count   | int: Total number of row list elements   |
+|12 |  32  | reserved| padding                                  |
+
+> TODO-LDML: Should there be a '0-key2count terminator'?
+
+For each 'row' list element
+
+| ∆ | Bits | Name       | Description                              |
+|---|------|------------|------------------------------------------|
+|16+|  32  | key2       | int: index into `key2` element           |
+|24+|  32  | key2count  | int: count of `key2` elements            |
+
+### C7043.2.x `key2`—Secondary key list
+
+This is a list of vkeys, but ordered by rows.  The `rows` elements point into the start and length of entries here.
+
+> TODO-LDML: Should there be a '0-vkey terminator'?
+> And shold the 0th element also be a 0 vkey so that key2=0 means null?
+
+| ∆ | Bits | Name    | Description                              |
+|---|------|---------|------------------------------------------|
+| 0 |  32  | ident   | `key2`                                   |
+| 4 |  32  | size    | int: Length of section                   |
+| 8 |  32  | count   | int: Total number of key2 elements       |
+|12 |  32  | reserved| padding                                  |
+
+For each `key2` element:
+
+Note that the first two fields are the index into the `keys` element, except arranged according to the keyboard layout.
+
+| ∆ | Bits | Name    | Description                              |
+|---|------|---------|------------------------------------------|
+|16+|  32  | vkey    | int: vkey ID                             |
+|20+|  32  | mod     | int: modifier key flags                  |
```

Comment:
I don't think this is correct. This should be an array of `keys` indices, and have no reference either to `vkey` or to `mod`.

Reply:
that's a better idea.

Reply:
I'm a little unsure about how the modifier flags in here on the target `key` element.

Reply:
modifier flags are determined by the layer, aren't they?

Reply:
for hardware-yes.  good point, each key inherits the modifier flags from that layer. For non hardware, it's more complex.  This gets back to the discussion about what `layer id="shift"` denotes for a non hardware layer.

Reply:
Is modifier even relevant for non-hardware? The output is defined by the `<key id>` association.

Reply:
Agreed

Reply:
More later